src/main/scala/com/gu/rcspollerlambda/services/Kinesis.scala

```diff
@@ -0,0 +1,45 @@
+package com.gu.rcspollerlambda.services
+
+import java.util.UUID
+
+import com.amazonaws.services.kinesis.model.PutRecordRequest
+import com.amazonaws.services.kinesis.{ AmazonKinesis, AmazonKinesisClientBuilder }
+import com.gu.rcspollerlambda.config.Config
+import com.gu.rcspollerlambda.models.{ KinesisPublishError, LambdaError }
+import io.circe.generic.semiauto.deriveEncoder
+import io.circe.syntax._
+import io.circe.{ Encoder, Json, Printer }
+
+object Kinesis extends Config with Logging {
+
+  lazy val client: AmazonKinesis = AmazonKinesisClientBuilder.standard().
+    withRegion(AWS.awsRegion).
+    withCredentials(AWS.awsCredentials).
+    build()
+
+  def publish(subject: String, message: Json): Either[LambdaError, Unit] = {
+    val updateMessage = UpdateMessage(subject = subject, rcsUpdate = message)
+
+    val partitionKey = UUID.randomUUID().toString
+
+    implicit val encoder: Encoder[UpdateMessage] = deriveEncoder[UpdateMessage]
+    val printer = Printer.noSpaces.copy(dropNullValues = true)
+    val payload = printer.prettyByteBuffer(updateMessage.asJson)
```

Comment:
interesting, what does this do?

Reply:
This is serialize a case class to JSON.
So.... why you ask is that there.

1) Trying to get a strongly typed representation of the message format before it goes onto the wire. Which would allow the consumer to have more idea about what to expect.

2) Ultimately what to be able to change the method signitures on the ElasticSearch classes to say what objects they are looking for. At the moment it's end to end Json or JsValue which I found very difficult to reason about.

That's abit opinionated hence why it's isolated to the Kinesis stream so the team has the option of dropping it when I'm done.

Reply:
👍 

I was about to suggest a more familiar `Json.toJson` until I remembered that's a Play API 🙉 .

This makes sense.